unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,434 @@
+    def test_execute_batch_commands_clear_command_buffer_called(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.execute_batch_commands()
+        aggregator_explicit._client_command_buffer.clear.assert_called_once()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_execute_batch_command_methods", "mock_start_websocket_funcs")
+    def test_execute_batch_commands_wait_for_command_response_called(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.execute_batch_commands()
+        aggregator_explicit.dispatcher.wait_for_command_response.assert_called_once()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_execute_batch_command_methods", "mock_start_websocket_funcs")
+    def test_execute_batch_commands_returns_response(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.dispatcher.wait_for_command_response.return_value = \
+            TEST_BATCH_COMMAND_RESPONSE
+        response = aggregator_explicit.execute_batch_commands()
+        assert response == TEST_BATCH_COMMAND_RESPONSE
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_execute_batch_command_methods", "mock_start_websocket_funcs")
+    def test_execute_batch_commands_log_bid_offer_confirmation_called(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.log_bid_offer_confirmation") as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.dispatcher.wait_for_command_response.return_value = \
+                TEST_BATCH_COMMAND_RESPONSE
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_execute_batch_command_methods", "mock_start_websocket_funcs")
+    def test_execute_batch_commands_log_deleted_bid_offer_confirmation_called(
+            aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.log_deleted_bid_offer_confirmation"
+                   ) as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.dispatcher.wait_for_command_response.return_value = \
+                TEST_BATCH_COMMAND_RESPONSE
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called()
+
+
+def test_list_aggregators_request_empty_list(aggregator_explicit):
+    with patch("gsy_e_sdk.aggregator.blocking_get_request", return_value=None):
+        aggs_list = aggregator_explicit.list_aggregators()
+        assert aggs_list == []
+
+
+def test_list_aggregators_request_not_empty_list(aggregator_explicit):
+    with patch("gsy_e_sdk.aggregator.blocking_get_request", return_value=[TEST_AGG]):
+        aggs_list = aggregator_explicit.list_aggregators()
+        assert aggs_list == [TEST_AGG]
```

Comment:
I am aware of [this](https://github.com/gridsingularity/gsy-e-sdk/pull/194#discussion_r764776108) commentm however, why not parametrize the outcome of blocking_get_request and the expected outcome ?
```python
@pytest.mark.parametrize("blocking_request_ret_val, expected_ret_val", [[None, TEST_AGG], [[], TEST_AGG]])
```
?

Reply:
I did not noticed the first time how it could be achieved easily with parametrize. I therefore ended introducing conditions inside the test and that was not a good practice. 
However this time the way the parametrize could be implemented without different logic between each case seems direct as you point out. 
I'll address it.